hubbard/ncsile.py

```diff
@@ -1,35 +1,135 @@
 import numpy as np
 import sisl
+import warnings
 
 __all__ = ['ncSilehubbard']
 
 
+def get_sile(file, *args, **kwargs):
```

Comment:
Hi Nick, I'm having some problems when trying to implement the lines from above (mainly using `sisl` to add `ncsilehubbard` as a `sile` and then using the functions from `ncsilehubbard` to read the densities, U, etc).

I'm probably doing something wrong but I have found a problem when trying to read a file whose path is in another folder (when the file is in the working directory it runs normally).

For instance, when I run this peace of code:

```python
import sisl
from hubbard import HubbardHamiltonian, ncsile, sp2
import numpy as np

# Build sisl Geometry object only for a subset of atoms
molecule = sisl.Geometry([0,0,0], atoms=sisl.Atom(6), sc=[1,1,1])
molecule.sc.set_nsc([1, 1, 1])

# Build HubbardHamiltonian object
Hsp2 = sp2(molecule)
H = HubbardHamiltonian(Hsp2, U=3.5)
# Generate simple density
H.n = np.ones((2, H.sites))

# Create another folder
import os
os.mkdir('another_folder')

file_name = 'another_folder/test.nc'
H.write_density(file_name)

# If the file is not in the working directory then it crashes
sisl.io.add_sile(file_name, ncsile.ncSilehubbard, gzip=False)
HU = sisl.get_sile(file_name)
print('n: ', HU.read_density())
print('U: ', HU.read_U())
print('kT: ', HU.read_kT())
```

I get the following output:
```
  File "test-hubbard-2.py", line 23, in <module>
    print('n: ', HU.read_density())
  File "/home/sofia/anaconda3/lib/python3.8/site-packages/sisl-0.12.0.dev0-py3.8-linux-x86_64.egg/sisl/io/sile.py", line 463, in __getattr__
    return getattr(self.fh, name)
  File "src/netCDF4/_netCDF4.pyx", line 2945, in netCDF4._netCDF4.Dataset.__getattr__
  File "src/netCDF4/_netCDF4.pyx", line 2887, in netCDF4._netCDF4.Dataset.getncattr
  File "src/netCDF4/_netCDF4.pyx", line 1468, in netCDF4._netCDF4._get_att
  File "src/netCDF4/_netCDF4.pyx", line 1950, in netCDF4._netCDF4._ensure_nc_success
AttributeError: NetCDF: Attribute not found
```

If now I change `folder_name='test.nc'` then it works... I have no idea why this happens.

Reply:
Yeah this isn't the way the `add_sile` is supposed to work.

The idea is that the suffix of the file describes which handle it should read from. A generic `*.nc` file suffix will be understood as a Siesta-nc file. That is why I think you should add another suffix that describes the hubbard file-format. It could be `HU.nc`, or `U.nc` or what-ever?

Then `add_sile("HU.nc", ncSileHubbard)` will use the `ncSileHubbard` class to read content whenever `get_sile("*.HU.nc")` gets invoked? Does this make sense?

Reply:
Oh I see, I misunderstood the idea. Thanks!

I think it's fine to have that option. If the user saves the file with this suffix then it works (see commit d1a087f ) :)